mne/io/tag.py

```diff
@@ -203,6 +204,257 @@ def _loc_to_eeg_loc(loc):
+        elif matrix_type == FIFF.FIFFT_COMPLEX_FLOAT:
+            data = np.fromstring(read_big(fid, 4 * 2 * dims.prod()),
+                                 dtype='>f4')
+            # Note: we need the non-conjugate transpose here
+            data = (data[::2] + 1j * data[1::2])
+        elif matrix_type == FIFF.FIFFT_COMPLEX_DOUBLE:
+            data = np.fromstring(read_big(fid, 8 * 2 * dims.prod()),
+                                 dtype='>f8')
+            # Note: we need the non-conjugate transpose here
+            data = (data[::2] + 1j * data[1::2])
+        else:
+            raise Exception('Cannot handle matrix of type %d yet'
+                            % matrix_type)
+        data.shape = dims
+    elif matrix_coding in (_matrix_coding_CCS, _matrix_coding_RCS):
+        from scipy import sparse
+        # Find dimensions and return to the beginning of tag data
+        pos = fid.tell()
+        fid.seek(tag.size - 4, 1)
+        ndim = int(np.fromstring(fid.read(4), dtype='>i4'))
+        fid.seek(-(ndim + 2) * 4, 1)
+        dims = np.fromstring(fid.read(4 * (ndim + 1)), dtype='>i4')
+        if ndim != 2:
+            raise Exception('Only two-dimensional matrices are '
+                            'supported at this time')
+
+        # Back to where the data start
+        fid.seek(pos, 0)
+        nnz = int(dims[0])
+        nrow = int(dims[1])
+        ncol = int(dims[2])
+        sparse_data = np.fromstring(fid.read(4 * nnz), dtype='>f4')
+        shape = (dims[1], dims[2])
+        if matrix_coding == _matrix_coding_CCS:
+            #    CCS
+            tmp_indices = fid.read(4 * nnz)
+            sparse_indices = np.fromstring(tmp_indices, dtype='>i4')
+            tmp_ptrs = fid.read(4 * (ncol + 1))
+            sparse_ptrs = np.fromstring(tmp_ptrs, dtype='>i4')
+            if (sparse_ptrs[-1] > len(sparse_indices) or
+                    np.any(sparse_ptrs < 0)):
+                # There was a bug in MNE-C that caused some data to be
+                # stored without byte swapping
+                sparse_indices = np.concatenate(
+                    (np.fromstring(tmp_indices[:4 * (nrow + 1)], dtype='>i4'),
+                     np.fromstring(tmp_indices[4 * (nrow + 1):], dtype='<i4')))
+                sparse_ptrs = np.fromstring(tmp_ptrs, dtype='<i4')
+            data = sparse.csc_matrix((sparse_data, sparse_indices,
+                                     sparse_ptrs), shape=shape)
+        else:
+            #    RCS
+            sparse_indices = np.fromstring(fid.read(4 * nnz), dtype='>i4')
+            sparse_ptrs = np.fromstring(fid.read(4 * (nrow + 1)), dtype='>i4')
+            data = sparse.csr_matrix((sparse_data, sparse_indices,
+                                     sparse_ptrs), shape=shape)
+    else:
+        raise Exception('Cannot handle other than dense or sparse '
+                        'matrices yet')
+    return data
+
+
+def _read_simple(fid, tag, shape, rlims, dtype):
```

Comment:
lol bureaucrats ;) My main objection was that we should be careful of making PRs which are hard to review. Eric posted a PR which said that it made IO faster. There was one line in the diff which claimed to make something faster. Was that it? I'm still in the dark as to what made it faster. Anyway, I trust Eric with not introducing bugs so no issues :)